### DISCUSSION TITLE
Hard filters

### DISCUSSION THREAD

**Discussion by pogran** [prisma/prisma#17178] [Q&A] [answered]
How to get data for filters. for example I have table books (id, title) and table genres (id, bookId, title). its one to many.

I want to create request . get all books witch have all books_genres from list (genres.id)? i think need to use findMany, but how create where condition . or its sth with having and groupBy

**Comment by pogran** [comment]
for example its clear mysql witch work perfect in my case 

SELECT b.*, COUNT(g.genreId) c FROM books b 
	LEFT JOIN books_genres g
	ON b.id = g.bookId
	WHERE g.genreId IN (2, 3, 6)
	GROUP BY b.id
	HAVING c = 3
	ORDER BY b.id DESC
	LIMIT 20

but how it write in prisma language...

**Comment by pogran** [comment, marked as answer]
solve this question, mb its help sb

`
const genreGroups = await prisma.bookGenre.groupBy({
			by: ['bookId'],
			where: {
				genreId: {
					in: genres
				},
			},
			orderBy: {
				bookId: 'desc'
			},
			having: {
				bookId: {
					_count: {
						equals: genres.length,
					}
				}
			},
			take: countPerPage,
		});
		return genreGroups.map(b => b.bookId);
`

and after this send books.findMany with this ids

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
